### ISSUE TITLE
Unexpected behavior when updating Ticket ID in a Beacon Issue

### ISSUE THREAD

**Issue by lukmay** [noi-techpark/it.bz.beacon.admin#106]
When updating the Ticket ID after it already was set once, I experienced weird behavior. 
Not a singe mail from changes on this issue where send anymore and after creating a new issue on the same beacon the Ticket ID I used to update the other ticket before got set as the Ticket ID in the newly created issue.

**Comment by noctho** [comment]
> Not a singe mail from changes on this issue where send anymore and after creating a new issue

What do you mean exactly? Did you only not get the mails after updates on the issue itself (title, description or ticketId), or also the mails about comments on the issue?

The mail notification is implemented this way: A mail is sent only after
 * a new issue is created
 *  when a comment is created on the issue
 *  when the issue was closed / reopened.

In case of a comment, the writer of the comment does not receive the mail, but only the reporter of the issue and all others who have commented the issue so far. Every mail will be also sent to the support address. This was specified with @sseppi to avoid sending too much spam. After today's meeting with @sseppi we have specified that when an update is made on the issue or a comment, a mail will be sent to the support address (for the internal issue tracker).

Did you notice any anomaly from the previous approach?

> after creating a new issue on the same beacon the Ticket ID I used to update the other ticket before got set as the Ticket ID in the newly created issue.

Can you please provide more information about this bug (Screenshots, Beacon Ids,  ect.)? Unfortunately I was not able to reproduce it?
This was my approach to reproduce the bug:
1. I created Issue1.
2. I updated the TicketId of Issue1.
3. I created Issue2 on the same Beacon.
4. Afterwards, Issue1 had assigned the entered TicketId and Issue2 had no TicketId assigned.

**Comment by noctho** [comment]
@lukmay Could it be that the old issue was still visible after you created the new issue? In the version you tested, it was implemented in that way that it was not being displayed after the issue was created. If you had another issue open before, this one was still displayed. This is a bit confusing behavior. In the new PR (https://github.com/noi-techpark/it.bz.beacon.admin/pull/107) I have changed the behavior so that after the successful creation of an issue it will always be shown.

**Comment by lukmay** [comment]
Hi @noctho, Sorry for the late response.
The problem I described does still exist.
When just creating a ticket and setting a TicketId, everything works fine.
The issue only accrues after updating an issue that already had a TicketId. 
So the Steps to reproduce it would be the following:

1. created Issue1.
2. updated the TicketId of Issue1.
3. **set the TicketId to another TicketId** 
4. created Issue2 on the same Beacon.
5. Afterwards, Issue1 had assigned the entered TicketId and Issue2 had no TicketId assigned.

**Comment by noctho** [comment]
Hi @lukmay, if I understood your steps correctly, at 3 you update the TicketId of Issue1. Then at 5 Issue2 has no TicketId assigned (which is the correct behavior) and Issue1 had the updated or di old TicketId assigned?

I tried reproducing the steps you described and after step 5 Issue1 had the updated TicketId assigned. That would be exactly the behavior that was implemented.

**Comment by lukmay** [comment]
@noctho oh I made a mistake in the Steps.. When I tried it step 5 was that Issue1 had the updated TicketId bud doesn't sent any new mails and Issue2 also has the same TicketId.

If you want, I can show you exactly what I mean in a short discord or google meets call.

**Comment by noctho** [comment]
The mail after the update of the TickedId is sent only to the Beacon support address. It will not be sent to the reporter or the commentators.

It would probably be better to discuss the problem with the TicketId it in a call. When do you have time for this and where can I contact you?

**Comment by lukmay** [comment]
Great @noctho. I am available all time today until 4pm.
Whenever it's best for you, you could join this Discord Server https://discord.gg/hKSXk9eZ and ping me (Lukas^^#0087) 
or send me a meeting link via GitHub.

**Comment by noctho** [comment]
I tested again the whole workflow and noticed the following sent mails:

1. Create Issue 1 on beacon with the name 'Kontakt'
> TO: support@beacon, \<reporter-mail-address\>
> Subject: [issues.opendatahub.bz.it #] New issue for beacon Kontakt
>
> Content:
> A new issue has been reported by **tnocker** for beacon **Kontakt**:
> ### Issue1
> \-\-\-
> View Issue 

2. Assign 123 as TicketId for Issue1
> TO: support@beacon
> Subject: [issues.opendatahub.bz.it #123] New Ticket-Id for issue
>
> Content:
> The Ticket-Id of an issue has been updated to #123 by **tnocker** for beacon **Kontakt**:
> ### Issue1
> \-\-\-
> View Issue 

3. Update the TicketId of Issue1 to 1234
> TO: support@beacon
> Subject: [issues.opendatahub.bz.it #1234] New Ticket-Id for issue
>
> Content:
> The Ticket-Id of an issue has been updated from #123 to #1234 by **tnocker** for beacon **Kontakt**:
> ### Issue1
> \-\-\-
> View Issue 

4. Comment 'test' with `anotheruser` on Issue 1
> TO: support@beacon, \<reporter-mail-address\>
> Subject: [issues.opendatahub.bz.it #1234] Issue1
>
> Content:
> **anotheruser** commented:
>
> test
> \-\-\-
> View Issue 

5. Creating Issue 2 on the same beacon:
> TO: support@beacon, \<reporter-mail-address\>
> Subject: [issues.opendatahub.bz.it #] New issue for beacon Kontakt
>
> Content:
> A new issue has been reported by **tnocker** for beacon **Kontakt**:
> ### Issue2
> \-\-\-
> View Issue 

6. Comment 'test' with `anotheruser` on Issue 2
> TO: support@beacon, <reporter-mail-address>
> Subject: [issues.opendatahub.bz.it #] Issue2
>
> Content:
> **anotheruser** commented:
>
> test
> \-\-\-
> View Issue 

@lukmay Can you please check if you receive the same mails with these steps, or if any of them are missing? I could also modify one of these mails if you want a different subject or content. E.g. it is possible to add the id of the BeaconAPI to handle issues without a TicketId more easily.

NB: For steps 4 and 6, in general, all other commentators except anotheruser will receive this mail. In this example, however, that is not the case because there are no other commentators.

**Comment by lukmay** [comment]
The Content of the Mails is perfect!
Also, creating a new Issue no matter what happened with the previous one works flawless.

The only Problem now is that still after changing the Ticket-Id (Step 3 in your example) there are no more mails send from this Beacon Issue.
(No mail when commenting, closing.. ) 

Is there a way to check the logs if a mail was sent in this case from the Beacon Admin Aplication to the support@beacon or not?

**Comment by noctho** [comment]
@lukmay I have created a PR with which each single mail is logged (https://github.com/noi-techpark/it.bz.beacon.api/pull/84). Once this is merged into the main branch, the logs can be checked.
